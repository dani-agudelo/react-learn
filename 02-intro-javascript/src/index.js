import {heroes} from './data/heroes';


const getHeroById = (id) => {
    return heroes.find(hero => hero.id === id);
}

const heroe= getHeroById(2);
console.log(heroe); // { id: 2, name: 'Spiderman', owner: 'Marvel' }

const getHeroesByOwner = (owner) => {
    return heroes.filter(hero => hero.owner === owner);
}
const heroesByDC = getHeroesByOwner('DC');
console.log(heroesByDC); // [{ id: 1, name: 'Batman', owner: 'DC' }, { id: 3, name: 'Superman', owner: 'DC' }, { id: 4, name: 'Flash', owner: 'DC' }]