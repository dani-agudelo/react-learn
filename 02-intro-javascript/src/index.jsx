import React from 'react';
import ReactDOM from 'react-dom/client';
import ArmaTourism from './main'; // Aseg√∫rate de que la ruta sea correcta
import './index.css'; // Si tienes estilos globales

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <ArmaTourism />
  </React.StrictMode>
);