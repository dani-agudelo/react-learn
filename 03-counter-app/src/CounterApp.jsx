import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {

    // useState es un Hook que permite añadir el estado de React a un componente funcional.
    // useState devuelve un array con dos elementos: el estado actual y una función para actualizar

    const [counter, setCounter] = useState(value);

    const handleAdd = () => setCounter(counter + 1);

    const handleSubtract = () => setCounter(counter - 1);

    const handleReset = () => setCounter(value);


    return (
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>

            <button onClick={handleAdd}> +1 </button>
            <button onClick={handleSubtract}> -1 </button>
            <button onClick={handleReset}> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};