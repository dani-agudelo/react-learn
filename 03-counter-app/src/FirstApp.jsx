import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle }) => {
  return (
    <>
      <h1>{ title }</h1>
      <p>{ subtitle }</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string
};

FirstApp.defaultProps = {
  title: 'No hay título',
  subtitle: 'No hay subtítulo',
};