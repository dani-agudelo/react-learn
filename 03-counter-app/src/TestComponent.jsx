import PropTypes from 'prop-types';

export const TestComponent = ({ message }) => {
  return <h1>{message}</h1>;
};

TestComponent.propTypes = {
  message: PropTypes.string,
};

TestComponent.defaultProps = {
  message: 'Mensaje por defecto',
};