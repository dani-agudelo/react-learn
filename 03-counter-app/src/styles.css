html,
body {
  background-color: #21232a;
  color: white;
  font-family: "Helvetica", Arial, sans-serif;
  font-size: 1.3 rem;
  padding: 70px;
}


button {
    background-color: #3a3f47;
    border: none;
    border-radius: 5px;
    color: white;
    cursor: pointer;
    font-size: 1.2rem;
    padding: 10px 20px;
}

button:hover {
    background-color: #4a4f57;
}