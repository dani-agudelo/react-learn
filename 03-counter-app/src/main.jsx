import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import { CounterApp } from './CounterApp'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value= {0   } />
    </React.StrictMode>,
)


// Para tener una app de React, básicamente necesitas:
// 1. Un elemento HTML con un id específico (en este caso, 'root') en tu archivo HTML.
// 2. Importar React y ReactDOM en tu archivo JavaScript.
// 3. Crear un componente React (en este caso, App) que renderice algo.
// 4. Usar ReactDOM para renderizar ese componente dentro del elemento HTML especificado