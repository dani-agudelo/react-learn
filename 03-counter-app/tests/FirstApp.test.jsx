import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en first app component', () => {
  test('Debe hacer match con el snapshot', () => {

    const {container}= render(<FirstApp title="Hola, soy Vegeta" />);
    console.log(container);
    expect(container).toMatchSnapshot(); 
  });


})