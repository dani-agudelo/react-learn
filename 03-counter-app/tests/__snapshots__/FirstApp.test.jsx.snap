// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en first app component Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Vegeta
  </h1>
  <p />
</div>
`;
